sim/rogue/subtlety_rotation.go

```diff
@@ -0,0 +1,335 @@
+package rogue
+
+import (
+	"time"
+
+	"github.com/wowsims/wotlk/sim/core"
+	"github.com/wowsims/wotlk/sim/core/proto"
+)
+
+type subtletyPrio struct {
+	check GetAction
+	cast  DoAction
+	cost  float64
+}
+
+func (rogue *Rogue) setSubtletyBuilder() {
+	mhDagger := rogue.Equip[proto.ItemSlot_ItemSlotMainHand].WeaponType == proto.WeaponType_WeaponTypeDagger
+	// Garrote
```

Comment:
Mechanics remark: Premeditation is can also be cast while Shadow Dance is active (and has no GCD, so it can be macroed into abilities).

Reply:
Anything regarding rotation optimization is going to be dealt with later